src/melee/it/items/itdrmariopill.c

```diff
@@ -285,6 +355,97 @@ static s32 it_802C0E48(Item_GObj* gobj)
     return false;
 }
 
+static void it_802C0B5C_no_inline(Item_GObj* gobj)
+{
+    it_802C0B5C(gobj);
+}
+
+void it_802C0510(Item_GObj* parent_gobj, Vec3* pos, s32 arg2, ItemKind kind,
+                 f32 facing_dir)
+{
+    SpawnItem spawn;
+    Item_GObj* gobj;
+
+    spawn.kind = kind;
+    spawn.prev_pos = *pos;
+    spawn.prev_pos.z = 0.0f;
+    it_8026BB68(parent_gobj, &spawn.pos);
+    spawn.facing_dir = facing_dir;
+    spawn.x3C_damage = 0;
+    spawn.vel.x = spawn.vel.y = spawn.vel.z = 0.0f;
+    spawn.x0_parent_gobj = parent_gobj;
+    spawn.x4_parent_gobj2 = spawn.x0_parent_gobj;
+    spawn.x44_flag.b0 = 1;
+    spawn.x40 = 0;
+    gobj = Item_80268B18(&spawn);
+    if (gobj != NULL) {
+        it_802C0B5C_no_inline(gobj);
+        it_80273670(gobj, 0, (f32) arg2);
+        db_80225DD8(gobj, parent_gobj);
+        it_802750F8(gobj);
+        it_8026B3A8(gobj);
+    }
+}
+
+static void it_802C0DF8_no_inline(Item_GObj* gobj)
+{
+    it_802C0DF8(gobj);
+}
+
+static void it_802C0DF8_no_inline2(Item_GObj* gobj)
+{
+    it_802C0DF8_no_inline(gobj);
+}
+
+static void it_802C0DF8_no_inline3(Item_GObj* gobj)
+{
+    it_802C0DF8_no_inline2(gobj);
+}
```

Comment:
surely there is a less insane way to do this...?

Reply:
Yeah, by finding the original call chain. Maybe it's calling something from Mario fireball or elsewhere, which is calling something else.